src/apps/lwaftr/lwaftr.lua

```diff
@@ -606,7 +606,7 @@ local function from_b4(lwstate, pkt)
          end
          -- Fragment if necessary
          if pkt.length > lwstate.ipv4_mtu then
-            local fragstatus, frags = fragmentv4.fragment_ipv4(pkt, lwstate.ipv4_mtu)
+            local fragstatus, frags = fragmentv4.fragment_ipv4(pkt, constants.ethernet_header_size, lwstate.ipv4_mtu)
```

Comment:
The VLAN tagging in `lwaftr_nutmeg` is still using the kludge of modifying `constants.ethernet_header_size` so with VLAN tagging enabled, the correct value is gonna be passed here. The commit that adds the `l2_size` parameter to the `fragment_ipv4()` function allows to write the tests more easily and with (almost) the same code as the cherry-picked commit from `lwaftr_olive`. Plus, this makes the code of both versions closer, which makes backporting further patches more easily (if needed in the future), and at the same time the change is small enough that it won't unadvertedly break anything else.